src/controller/alunosController.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import { AlunosService } from '../services/alunosService';
+
+export class AlunosController {
+  private alunosService = new AlunosService();
+
+  /**
+   * Cria um novo aluno com os dados fornecidos no corpo da requisição.
+   *
+   * @param req - A requisição contendo os dados do novo aluno.
+   * @param res - A resposta a ser enviada ao cliente.
+   */
+  async criarAluno(req: Request, res: Response) {
+    try {
+      const { nomeCompleto, rg, numeroMatricula, turmaId, responsavelCpf } =
+        req.body;
+      const novoAluno = await this.alunosService.criarAluno(
+        nomeCompleto,
+        rg,
+        numeroMatricula,
+        Number(turmaId),
+        responsavelCpf
+      );
+
+      return res.status(201).json({
+        sucess: true,
+        message: 'Estudante criado com sucesso',
+      });
+    } catch (error) {
+      if (error.name === 'Conflito') {
+        return res.status(409).json({ message: error.message });
+      }
+      return res.status(400).json({ message: 'Erro ao criar aluno', error });
+    }
+  }
+
+  /**
+   * Lista todos os alunos, incluindo suas turmas, membros e responsáveis.
+   *
+   * @param req - A requisição do cliente.
+   * @param res - A resposta a ser enviada ao cliente com a lista de alunos.
+   */
+  async listarAlunos(req: Request, res: Response) {
+    try {
+      const alunos = await this.alunosService.listarAlunos();
+      res.json(alunos);
+    } catch (error) {
+      res.status(500).json({ message: 'Erro ao listar alunos', error });
+    }
+  }
+
+  /**
+   * Busca um aluno pelo ID fornecido nos parâmetros da requisição.
+   *
+   * @param req - A requisição do cliente contendo o ID do aluno.
+   * @param res - A resposta a ser enviada ao cliente com o aluno encontrado.
+   */
+  async buscarAlunosPorId(req: Request, res: Response) {
+    try {
+      const { id } = req.params;
+
+      const aluno = await this.alunosService.buscarAlunoPorId(Number(id));
+      return res.json(aluno);
+    } catch (error) {
+      return res.status(500).json({ message: 'Erro ao buscar aluno', error });
```

Comment:
Quando o aluno não for encontrado, é legal retornar um 404, e deixar o 500 para erros recuperáveis do servidor. Sugestão de leitura para entender melhor casos de uso para tratamentos de erro em Typescript: https://medium.com/with-orus/the-5-commandments-of-clean-error-handling-in-typescript-93a9cbdf1af5